### PR TITLE
Split codec_impl_fuzz_test targets out by protocol

### PR DESCRIPTION
**Pull request by KBaichoo**
This allows the fuzzer to focus on the individual protocols better and allows us to categorize failures by protocol.

Signed-off-by: Kevin Baichoo <kbaichoo@google.com>

Commit Message: Split codec_impl_fuzz_test targets out by protocol.
Additional Description:
Risk Level: low
Testing: built and ran target
Docs Changes: na
Release Notes: na
Platform Specific Features: na
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by adisuissa** [comment]
The downside of this approach is that now we have 4 different fuzzers that may benefit from running the same test across different instances (well... at least for the H/2 case).
I guess one can look at the stack trace and see the line number of the failing call to `codecFuzz()`, but this may change over-time.
What do you think about explicit functions that call the `codecFuzz()` functions, for example:
```
void codecFuzzHttp1() {
  codecFuzz(input, HttpVersion::Http1);
}
```

**Comment by KBaichoo** [comment]
Split out build target by protocol instead and added wrapper functions for the individual codecs as suggested.

File: test/common/http/codec_impl_fuzz_test.cc [hunk 1]
```diff
@@ -779,23 +779,47 @@ void codecFuzz(const test::common::http::CodecImplFuzzTestCase& input, HttpVersi
...
   try {
     // Validate input early.
     TestUtility::validate(input);
-    codecFuzz(input, HttpVersion::Http1);
-    codecFuzz(input, HttpVersion::Http2Nghttp2);
-    codecFuzz(input, HttpVersion::Http2WrappedNghttp2);
+#ifdef FUZZ_PROTOCOL_http1
+    codecFuzzHttp1(input);
+#endif
+#ifdef FUZZ_PROTOCOL_http2
```
Comment (adisuissa) [inline test/common/http/codec_impl_fuzz_test.cc:812]: nit: add a small comment explaining that we have different function names so they will appear explicitly in the stacktrace.

**Review by adisuissa — COMMENTED** [review]
Small nit, but otherwise LGTM.

File: test/common/http/codec_impl_fuzz_test.cc (hunk 1 above)
Comment (KBaichoo) [reply test/common/http/codec_impl_fuzz_test.cc:812]: done

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**